hugolib/pagination.go

```diff
@@ -414,6 +418,39 @@ func paginatePages(td targetPathDescriptor, seq interface{}, pagerSize int) (pag
 	return pagers, nil
 }
 
+func toPagesGroup(seq interface{}) (PagesGroup, error) {
+	if seq == nil {
+		return nil, nil
+	}
+
+	switch seq.(type) {
+	case PagesGroup:
+		return seq.(PagesGroup), nil
+	case []PageGroup:
+		return PagesGroup(seq.([]PageGroup)), nil
+	case []interface{}:
+		list := seq.([]interface{})
+		l := len(list)
+		if l == 0 {
+			break
+		}
+		switch list[0].(type) {
```

Comment:
`switch v := list[0].(type)` same comment as above -- avoids casting inside the switches.

Reply:
Not done as the value of the first element in the array is not important. Perhaps the check is not the good one.
There is a loop just after, iterating over all elements, to cast them one by one. The initial switch is here to handle type errors but perhaps it should be done for all elements.

If there is a better way to do it in go, tell me or fix it directly.